ibis/backends/dask/execution/window.py

```diff
@@ -0,0 +1,86 @@
+"""Code for computing window functions in the dask backend."""
+
+from typing import Any, Optional, Union
+
+import dask.dataframe as dd
+
+import ibis.expr.operations as ops
+import ibis.expr.window as win
+from ibis.expr.scope import Scope
+from ibis.expr.typing import TimeContext
+
+from ..core import execute_with_scope
+from ..dispatch import execute_node
+from .util import (
+    _pandas_dtype_from_dd_scalar,
+    _wrap_dd_scalar,
+    add_partitioned_sorted_column,
+    make_meta_series,
+)
+
+
+def _post_process_empty(
+    result: Any,
+    parent: Union[dd.Series, dd.DataFrame],
+    timecontext: Optional[TimeContext],
+) -> dd.Series:
+    """Post process non grouped, non ordered windows.
+
+    dd.Series/dd.DataFrame objects are passed through, otherwise we conform
+    the output to the parent input (i.e. so the shape an partitioning matches).
+
+    dd.core.Scalar needs special handling so downstream functions can work
+    with it.
+    """
+    if isinstance(result, (dd.Series, dd.DataFrame)):
+        return result
+    elif isinstance(result, dd.core.Scalar):
```

Comment:
https://github.com/dask/dask/blob/main/dask/dataframe/core.py#L115. It is a core dask dataframe type. 

Some things may result in a scalar `dd.col_name.mean()`.

Reply:
I see, it's Scalar value as a result of a distributed computation.

Reply:
Hmm..reading this more, why do we want to add the index for a scalar value? Seems unnecessarily but maybe I am missing sth..

Reply:
We project a scalar value to the full index, add the same partitioning scheme as the "parent" and return a `dd.Series` that can be concatted with all the other `selections`.

Reply:
KK. I think it might be worthy while add some comments here like

"
Eagerly compute the aggregation result, broadcast to match the parent shape and partitioning, finally adding index column so it can be concated back
"

"